### PR TITLE
refactor: Deduplicate get_new_slug method

### LINKED ISSUE #6984: Deduplicate get_new_slug method
`get_new_slug` method is duplicated many times through the project

### PR DESCRIPTION
**Pull request by sanjana-302**
Fixes #6984 

#### Short description of what this resolves:
Adds a generic get_new_slug method, prevents repetition of code.

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/models/event_topic.py  -1
+ app/models/event_sub_topic.py  -1
+ app/models/event_type.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297&bid=18144079)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/models/event_type.py  -1
+ app/models/event_topic.py  -1
+ app/models/event_sub_topic.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297&bid=18144079)

File: app/api/helpers/db.py [hunk 1]
```diff
@@ -115,3 +116,23 @@ def get_count(query):
     count_q = query.statement.with_only_columns([func.count()]).order_by(None)
     count = query.session.execute(count_q).scalar()
     return count
+
+
+def get_new_slug(name, model):
```
Comment (iamareebjamal) [inline app/api/helpers/db.py:121]: Great job, but this should take `(model, name)`, so it can be called like `get_new_slug(EventLocation, name)`

Comment (sanjana-302) [reply app/api/helpers/db.py:121]: Okay.I will do it.

File: app/models/event_topic.py [hunk 2]
```diff
@@ -1,25 +1,8 @@
-import uuid
-
-from app.api.helpers.db import get_count
+from app.api.helpers.db import get_count, get_new_slug
```
Comment (niranjan94) [inline app/models/event_topic.py:1]: Codacy Issue found: [Unused get_count imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity decreasing per file
==============================
+ app/models/event_type.py  -1
+ app/models/event_topic.py  -1
+ app/models/event_location.py  -1
+ app/models/event_sub_topic.py  -1
         

Clones removed
==============
+ app/models/event_type.py  -1
+ app/models/event_topic.py  -1
+ app/models/event_sub_topic.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297&bid=18144079)

File: app/models/event_sub_topic.py [hunk 3]
```diff
@@ -1,26 +1,8 @@
-import uuid
-
-from app.api.helpers.db import get_count
+from app.api.helpers.db import get_count, get_new_slug
```
Comment (niranjan94) [inline app/models/event_sub_topic.py:1]: Codacy Issue found: [Unused get_count imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297)

File: app/models/event_location.py [hunk 4]
```diff
@@ -1,24 +1,7 @@
-import uuid
-
-from app.api.helpers.db import get_count
+from app.api.helpers.db import get_count,  get_new_slug
```
Comment (niranjan94) [inline app/models/event_location.py:1]: Codacy Issue found: [Unused get_count imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297)

File: app/models/event_type.py [hunk 5]
```diff
@@ -1,25 +1,8 @@
-import uuid
-
-from app.api.helpers.db import get_count
+from app.api.helpers.db import get_count, get_new_slug
```
Comment (niranjan94) [inline app/models/event_type.py:1]: Codacy Issue found: [Unused get_count imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/models/event_topic.py  -1
+ app/models/event_location.py  -1
+ app/models/event_sub_topic.py  -1
+ app/models/event_type.py  -1
         

Clones removed
==============
+ app/models/event_topic.py  -1
+ app/models/event_sub_topic.py  -1
+ app/models/event_type.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297&bid=18144079)

File: app/models/event_location.py [hunk 6]
```diff
@@ -30,7 +13,7 @@ class EventLocation(db.Model):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.slug = get_new_slug(name=self.name)
+        self.slug = get_new_slug(EventLocation, name=self.name)
```
Comment (iamareebjamal) [inline app/models/event_location.py:16]:
```suggestion
        self.slug = get_new_slug(EventLocation, self.name)
```

File: app/models/event_sub_topic.py [hunk 7]
```diff
@@ -37,7 +19,7 @@ class EventSubTopic(SoftDeletionModel):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.slug = get_new_slug(name=self.name)
+        self.slug = get_new_slug(EventSubTopic, name=self.name)
```
Comment (iamareebjamal) [inline app/models/event_sub_topic.py:22]: Same. Remove keyword argument

File: app/models/event_topic.py [hunk 8]
```diff
@@ -34,7 +17,7 @@ class EventTopic(SoftDeletionModel):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.slug = get_new_slug(name=self.name)
+        self.slug = get_new_slug(EventTopic, name=self.name)
```
Comment (iamareebjamal) [inline app/models/event_topic.py:20]: Same

File: app/models/event_type.py [hunk 9]
```diff
@@ -32,7 +15,7 @@ class EventType(SoftDeletionModel):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.slug = get_new_slug(name=self.name)
+        self.slug = get_new_slug(EventType, name=self.name)
```
Comment (iamareebjamal) [inline app/models/event_type.py:18]: Same

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/models/event_location.py  -1
+ app/models/event_type.py  -1
+ app/models/event_sub_topic.py  -1
+ app/models/event_topic.py  -1
         

Clones removed
==============
+ app/models/event_type.py  -1
+ app/models/event_sub_topic.py  -1
+ app/models/event_topic.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5535297&bid=18144079)

**Pull request merged**